### ISSUE TITLE
Not showing right background on all monitors; unresponsive, runs out of gpu memory with cosmic-bg not running

### ISSUE THREAD

**Issue by ids1024** [pop-os/cosmic-workspaces-epoch#44]
I've noticed a weird slowness and crash, and it seems the issue was I didn't have cosmic-bg running...

Two issues here:
* The way backgrounds are handled requires `cosmic-bg` to be running to have `~/.local/state/cosmic/com.system76.CosmicBackground` in a correct state. Not necessarily a problem to require cosmic-bg, but it should probably be handled more gracefully?
* The fact this causes a crash is presumably a `iced_wgpu` and/or `wgpu` issue, like https://github.com/pop-os/cosmic-workspaces-epoch/issues/40. But it's a weird thing to see...
  - `gdb` shows it's allocating wgpu buffers to upload image to the atlas. Not sure why caching isn't working, and only isn't working in this case?
  - buffer being leaked might be `wgpu` issue

**Comment by ids1024** [comment]
I'm also seeing this with the `no-subsurfaces` feature. Not sure what made this start happening. That used to work.

**Comment by ids1024** [comment]
https://github.com/iced-rs/iced/pull/2425 sounds related.

**Comment by WatchMkr** [comment]
I noticed performance becomes really poor when wallpapers are set separately for each display.

**Comment by ids1024** [comment]
Yeah, I think that's the same issue. When I saw this, one output had the right background. I guess I hadn't had external monitors connected when I last ran `cosmic-bg` at the time.

 Probably if we apply https://github.com/iced-rs/iced/pull/2425 and resolve conflicts (or update our Iced) it will fixed it.

If the same image is used for all monitors, it's fine. But otherwise, it's being evicted from the cache when the next monitor is rendered. So it has to uploaded to GPU every frame.

The fact I saw this running out of GPU memory, though, suggests wgpu or iced_wgpu is also leaking buffers somewhere...
